src/ConnectedMode/Suppressions/ServerIssuesStore.cs

```diff
@@ -19,24 +19,59 @@
  */
 
 using System;
-using System.Linq;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Linq;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.ConnectedMode.Suppressions
 {
     [Export(typeof(IServerIssuesStore))]
+    [Export(typeof(IServerIssuesStoreWriter))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class ServerIssuesStore : IServerIssuesStore
+    internal class ServerIssuesStore : IServerIssuesStore, IServerIssuesStoreWriter
     {
-        private IEnumerable<SonarQubeIssue> serverIssues;
+        private List<SonarQubeIssue> serverIssues;
 
         public event EventHandler ServerIssuesChanged;
 
+        #region IServerIssuesStore implementation
+
         public IEnumerable<SonarQubeIssue> Get()
         {
             return serverIssues ?? Enumerable.Empty<SonarQubeIssue>();
         }
+
+        #endregion IServerIssuesStore implementation
+
+        #region IServerIssueStoreWriter implementation
+
+        public void AddIssues(IEnumerable<SonarQubeIssue> issues, bool clearAllExistingIssues)
+        {
+            if (clearAllExistingIssues || serverIssues == null)
```

Comment:
Not a massive fan of the serverIssues == null here however I wasn't sure where else to initialize it.

Reply:
You could have initialised as part of the definition(
```
private List<SonarQubeIssue> serverIssues = new List<SonarQubeIssue>();
```

That would also have allowed you to skip the `null` handling in the `Get` method.
However, it doesn't make too much difference.